state/accountsDB.go

```diff
@@ -1145,8 +1154,50 @@ func (adb *AccountsDB) CancelPrune(rootHash []byte, identifier TriePruningIdenti
 	adb.storagePruningManager.CancelPrune(rootHash, identifier, adb.getMainTrie().GetStorageManager())
 }
 
+func (adb *AccountsDB) waitForStorageEpochChange(args StorageEpochChangeWaitArgs) error {
+	if args.SnapshotWaitTimeout < args.WaitTimeForSnapshotEpochCheck {
+		return fmt.Errorf("timeout (%s) must be greater than wait time between snapshot epoch check (%s)", args.SnapshotWaitTimeout, args.WaitTimeForSnapshotEpochCheck)
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), args.SnapshotWaitTimeout)
+	defer cancel()
+
+	for {
+		storageManager := adb.getMainTrie().GetStorageManager()
+		latestStorageEpoch, err := storageManager.GetLatestStorageEpoch()
+		if err != nil {
+			return err
+		}
+
+		if latestStorageEpoch == args.Epoch {
+			return nil
+		}
+
+		if storageManager.IsClosed() {
```

Comment:
I think this will be better moved on L1167, immediately after getting the storageManager pointer

Reply:
Done.